src/base/bittorrent/session.cpp

```diff
@@ -4298,10 +4379,66 @@ void Session::handlePeerBlockedAlert(libt::peer_blocked_alert *p)
         Logger::instance()->addPeer(QString::fromLatin1(ip.c_str()), true, reason);
 }
 
+namespace
+{
+    // helpers to handle differences in libtorrent 1.1 and 1.2 APIs
+    template <class AlertType>
+    std::pair<libt::address, int> alertAddressAndPort(const AlertType* a)
```

Comment:
We only need to do this a few times. Why not just have two code branches in respective alert handlers? Then it will be easier for us to get rid of the libtorrent-1.1 support when the time comes.